code/drasil-code/lib/Language/Drasil/Code/Imperative/Descriptions.hs

```diff
@@ -40,7 +41,7 @@ unmodularDesc = do
       implTypeStr Program = "program"
       implTypeStr Library = "library"
   return $ show $ sentenceDoc (sysinfodb spec) Implementation OneLine $ capSent $
-    foldlSent ([S "a", S (implTypeStr (implType g)), S "to"] ++ purpose spec)
+    foldlSent ([S "a", S (implTypeStr (implType g)), S "to"] ++ codeSpec g ^. sysInfo .purpose)
```

Comment:
You should not need to "double lens" (i.e. `.sysInfo .purpose`). If you make the right instance, `.purpose` should be enough.